### PR TITLE
Support --allow-path in the CLI

### PR DESCRIPTION
**Pull request by axic**

### CODE REVIEW

File: solc/CommandLineInterface.cpp [hunk 1]
```diff
@@ -533,7 +535,12 @@ Allowed options)",
 			"Switch to linker mode, ignoring all options apart from --libraries "
 			"and modify binaries in place."
 		)
-		(g_argMetadataLiteral.c_str(), "Store referenced sources are literal data in the metadata output.");
+		(g_argMetadataLiteral.c_str(), "Store referenced sources are literal data in the metadata output.")
+		(
+			g_argAllowPath.c_str(),
+			po::value<string>()->value_name("path"),
```
Comment (chriseth) [inline solc/CommandLineInterface.cpp]: `path(s)`?

**Review by chriseth — CHANGES_REQUESTED** [review]
Perhaps call it `--allow-paths` and please add an explanation in the documentation.

**Comment by axic** [comment]
Changed. Please review the docs, don't know if the sentence is true there :)

File: Changelog.md [hunk 2]
```diff
@@ -2,6 +2,8 @@
 
 Features:
  * Support ``interface`` contracts.
+ * Commandline interface: Support ``--allow-paths`` to define trusted import paths. Note: the
+   path(s) of the supplied contract(s) is always trusted.
```
Comment (chriseth) [inline Changelog.md]: Should be `supplied source file(s)`.

File: docs/using-the-compiler.rst [hunk 3]
```diff
@@ -29,6 +29,8 @@ files reside, so things like ``import "/etc/passwd";`` only work if you add ``=/
 
 If there are multiple matches due to remappings, the one with the longest common prefix is selected.
 
+Important to note is that paths of included sources files and paths defined by remappings are allowed to include, but everything else is rejected. Additional paths can be added via the ``--allow-paths /sample/path,/another/sample/path`` switch.
```
Comment (chriseth) [inline docs/using-the-compiler.rst]: `paths of source files specified on the commandline and paths defined by remappings...`

It is probably better to put that in line 16.

Comment (axic) [reply docs/using-the-compiler.rst]: Reworded, still not sure. Where exactly at line 16 should it go?

**Review by chriseth — APPROVED** [review]

**Pull request merged**